R/utils-features.R

```diff
@@ -64,6 +64,16 @@ scoreFeatQuality <- function(quality, values)
 
 hasFGroupScores <- function(fGroups) nrow(groupScores(fGroups)) > 0
 
+getFilteredFGroups <- function(fGroups, retFilter) {
+    if (!is.null(retFilter$subset)){
+        fGroups <- fGroups[retFilter$subset$i, retFilter$subset$j]
```

Comment:
Maybe add something like:

```r
if (!is.null(...))
{
    if (!is.null(retFilter$subset[["i"]]))
        fGroups <- fGroups[retFilter$subset$i, retFilter$subset$j]
    else
        fGroups <- fGroups[, retFilter$subset$j]
}
```
Then in the filter functions, you can drop all the `i = TRUE` assignments.

PS: please put curly braces on a new line for consistency :-)

Reply:
And I guess you could do the same for `j`.

Reply:
okay, how about:
```suggestion
getFilteredFGroups <- function(fGroups, retFilter) 
{   
    if (!is.null(retFilter$subset))
    {
        if (is.null(retFilter$subset$i) & !is.null(retFilter$subset$j))
            fGroups <- fGroups[, retFilter$subset$j] 
        if (!is.null(retFilter$subset$i) & is.null(retFilter$subset$j))
            fGroups <- fGroups[retFilter$subset$i]
        else   
            fGroups <- fGroups[retFilter$subset$i, retFilter$subset$j]  
    }
    if (!is.null(retFilter$delete))
    {   
        if(!is.null(retFilter$delete$j))
            fGroups <- delete(fGroups, i = retFilter$delete$i, j = retFilter$delete$j)
    }
    return(fGroups)
}
```
P.S.: so it is preferred to not be explicit?

Reply:
Thanks!

> P.S.: so it is preferred to not be explicit?

Uhf it depends ... ;-) In general, explicit may be good, but there are limits when repeating the same many times. But in the this is something small and mostly a matter of style.